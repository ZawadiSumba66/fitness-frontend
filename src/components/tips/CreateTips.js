import { connect } from 'react-redux';
import { useState } from 'react';
// import PropTypes from 'prop-types';
import store from '../../store';
import { createTip } from '../../actions/tip_action';
import DashboardLeft from '../user/DashboardLeft';
import Flash from '../user/Flash';

const CreateTips = () => {
  const [title, setTitle] = useState('');
  const [description, setDescription] = useState('');
  const [benefits, setBenefits] = useState('');
  const [instructions, setInstructions] = useState('');
  const [image, setImage] = useState(null);

  const handleFileUpload = (files) => {
    setImage(files[0]);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    const tip = {
      title,
      description,
      benefits,
      instructions,
      image,
    };
    store.dispatch(createTip(tip));
  };

  return (
    <div>
      <DashboardLeft />
      <div className="user__info p-3">
        <div className="rounded p-3 mt-4">
          <Flash />
        </div>
      </div>
      <div className="create-tips">
        <form onSubmit={handleSubmit}>
          <div className="row mx-auto w-50 bg-white p-5">
            <h3>Create Fitness Tips</h3>
            <input
              value={title}
              onChange={(e) => setTitle(e.target.value)}
              type="text"
              name="title"
              placeholder="Enter the Title"
              className="form-control"
              rows="3"
            />
            <textarea
              value={description}
              onChange={(e) => setDescription(e.target.value)}
              name="description"
              placeholder="Enter a brief description"
              className="form-control mt-3"
              rows="3"
            />
            <textarea
              value={instructions}
              onChange={(e) => setInstructions(e.target.value)}
              name="instructions"
              placeholder="Add some instructions to follow"
              className="form-control mt-3"
              rows="3"
            />
            <textarea
              value={benefits}
              onChange={(e) => setBenefits(e.target.value)}
              name="benefits"
              placeholder="What are some of the benefits a user will gain"
              className="form-control mt-3"
              rows="3"
            />
            <div className="d-flex flex-column pt-2">
              <input
                onChange={(e) => handleFileUpload(e.target.files)}
                type="file"
                name="image"
              />
              <br />
              <input
                type="submit"
                className="button-orange text-light btn font-weight-bold"
                value="Submit"
              />
            </div>
          </div>
        </form>
      </div>
    </div>
  );
};

const mapDispatchToProps = (dispatch) => ({
  createtip: () => dispatch(createTip),
});

const mapStateToProps = (state) => ({
  error: state.tipsReducer.error,
});

// CreateTips.propTypes = {
//   createtip: PropTypes.func.isRequired,
//   // error: PropTypes.string.isRequired,
// };
export default connect(mapStateToProps, mapDispatchToProps)(CreateTips);
