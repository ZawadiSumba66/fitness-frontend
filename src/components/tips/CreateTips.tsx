import { connect } from 'react-redux';
import { useState } from 'react';
import store from '../../store';
import { createTip } from '../../actions/tip_action';
import DashboardLeft from '../user/DashboardLeft';
import Flash from '../user/Flash';

type TipsCreate = {
  error: string[]
};

const CreateTips: React.FunctionComponent<any> = ({ error }: TipsCreate) => {
  const [title, setTitle] = useState('');
  const [description, setDescription] = useState('');
  const [benefits, setBenefits] = useState('');
  const [instructions, setInstructions] = useState('');
  const [image, setImage] = useState(null);

  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    const tip = {
      title,
      description,
      benefits,
      instructions,
      image,
    };
    store.dispatch(createTip(tip));
    console.log(error);
  };

  const fileChange = (files: any) => {
    setImage(files[0]);
  };

  return (
    <div>
      <DashboardLeft />
      <div className="user__info p-3">
        <div className="rounded p-3 mt-4">
          <Flash />
          <div className="errors">
            {error ? (
              <div>
                {error.map((item: string) => (
                  <li className="text-danger" key={Date.now() * Math.random()}>{item}</li>
                ))}
              </div>
            ) : (
              ''
            )}
          </div>
        </div>
      </div>
      <div className="create-tips">
        <form onSubmit={handleSubmit}>
          <div className="bg-white p-5">
            <h3 className="pb-3 font-bold md:text-2xl text-xl text-orange">Create a Workout Tip</h3>
            <p className="font-weight-bold">Please fill all the fields below.</p>
            <br />
            <input
              value={title}
              onChange={(e) => setTitle(e.target.value)}
              type="text"
              name="title"
              placeholder="Enter the Title"
              className="form-control"
              data-rows="3"
            />
            <textarea
              value={description}
              onChange={(e) => setDescription(e.target.value)}
              name="description"
              placeholder="Enter a brief description"
              className="form-control mt-3"
              data-rows="3"
            />
            <textarea
              value={instructions}
              onChange={(e) => setInstructions(e.target.value)}
              name="instructions"
              placeholder="Add some instructions to follow"
              className="form-control mt-3"
              data-rows="3"
            />
            <textarea
              value={benefits}
              onChange={(e) => setBenefits(e.target.value)}
              name="benefits"
              placeholder="What are some of the benefits a user will gain"
              className="form-control mt-3"
              data-rows="3"
            />
            <label htmlFor="avatar">
              Choose an image
              <input
                type="file"
                className="w-100 my-3"
                onChange={(e) => fileChange(e.target.files)}
                id="avatar"
              />
            </label>
            <br />
            <input
              type="submit"
              className="button-orange mt-3 w-50 text-light btn font-weight-bold"
              value="Submit"
            />
          </div>
        </form>
      </div>
    </div>
  );
};

type TipError = {
  tipsReducer: {
    tip_error: string[]
  }
};
const mapStateToProps = (state: TipError) => ({
  error: state.tipsReducer.tip_error,
});

export default connect(mapStateToProps, null)(CreateTips);
