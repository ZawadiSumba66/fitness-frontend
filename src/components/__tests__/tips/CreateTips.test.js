import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../../store';
import CreateTips from '../../tips/CreateTips';

it('matches the create Tips snapshot', () => {
  const tree = renderer.create(<Provider store={store}><CreateTips /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
