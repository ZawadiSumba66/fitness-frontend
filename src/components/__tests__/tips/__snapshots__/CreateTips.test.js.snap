// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the create Tips snapshot 1`] = `
<div>
  <div
    className="d-flex justify-content-center align-items-center spinner__wrapper"
  >
    <div
      className="spinner-border text-primary"
      role="status"
    >
      <span
        className="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
  <div
    className="user__info p-3"
  >
    <div
      className="rounded p-3 mt-4"
    >
      <div />
    </div>
  </div>
  <div
    className="create-tips"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="row bg-white p-5"
      >
        <h3>
          Create a Fitness Tip
        </h3>
        <input
          className="form-control"
          name="title"
          onChange={[Function]}
          placeholder="Enter the Title"
          rows="3"
          type="text"
          value=""
        />
        <textarea
          className="form-control mt-3"
          name="description"
          onChange={[Function]}
          placeholder="Enter a brief description"
          rows="3"
          value=""
        />
        <textarea
          className="form-control mt-3"
          name="instructions"
          onChange={[Function]}
          placeholder="Add some instructions to follow"
          rows="3"
          value=""
        />
        <textarea
          className="form-control mt-3"
          name="benefits"
          onChange={[Function]}
          placeholder="What are some of the benefits a user will gain"
          rows="3"
          value=""
        />
        <input
          className="button-orange mt-3 w-50 text-light btn font-weight-bold"
          type="submit"
          value="Submit"
        />
      </div>
    </form>
  </div>
</div>
`;
